src/sender.rs

```diff
@@ -0,0 +1,104 @@
+use std::io::{self, Write};
+use std::net::{TcpListener, TcpStream};
+use std::sync::mpsc::{sync_channel, Receiver, SyncSender, TrySendError};
+use std::sync::{Arc, RwLock};
+use std::thread;
+
+use crate::errors::GeyserError;
+
+pub struct TcpSender {
+    conns: Arc<RwLock<Vec<SyncSender<Vec<u8>>>>>,
+}
+
+impl TcpSender {
+    pub fn default() -> Self {
+        TcpSender {
+            conns: Arc::new(RwLock::new(Vec::new())),
+        }
+    }
+
+    pub fn publish(&self, message: Vec<u8>) -> Result<(), GeyserError> {
+        let mut conns_to_remove = Vec::new();
+        let mut send_errs = 0;
+
+        {
+            let conns = self.conns.read().unwrap();
+            for (i, conn) in conns.iter().enumerate() {
+                if let Err(e) = conn.try_send(message.clone()) {
```

Comment:
maybe it's better to use map_err()?

Reply:
we need to continue sending to other sockets, if one failed, not return from func